optimize/types.go

```diff
@@ -26,10 +26,12 @@ const (
 	// NoOperation specifies that no evaluation or convergence check should
 	// take place.
 	NoOperation Operation = 0
+	// signal is used interally to signal completion.
+	signalOperation Operation = 1 << (iota - 1)
```

Comment:
This does not seem to be used. Apart from that, having non-exported operations would be a strange idea.

Reply:
It is used (as you probably see now). We can re-assess if it needs to be after I make the fixes you suggest below.

Reply:
Yes, sorry, it was in the diff for global.go which is hidden unless clicked.

Reply:
I think that for now we can leave it in, although after merging this PR I would like to return to this to see if it can be avoided. However, until then I would:
* rename it to `signalDone`
* move it to the end of Operations
* fix spelling of `internally` in the comment.